### DISCUSSION TITLE
Rad Pro: Multiple custom HV profiles

### DISCUSSION THREAD

**Discussion by ThatBum** [Gissio/radpro#168] [Ideas]
The Energy-saving profile is not strong enough to run the tube in my GC-01. I'm aware of the tuning procedure and have some 1G ohm resistors on order, but once dialed in I'd like the ability to switch between Accuracy and Energy-saving modes like some of the other models can. It'd be nice to select them from "HV profile" instead of having to remember the settings for one or the other and manually input them in the "Custom profile" menu. Thus, savable profiles.

Now I understand if you don't want this functionality on the device for fear of people messing with it and breaking things. Might I suggest adding it as functionality to `radpro-tool`? For example, these arguments would facilitate such functionality.

`radpro-tool.py --list-tube-hv-profile`
`radpro-tool.py --set-tube-hv-profile NAME FREQ DUTY_CYCLE`
`radpro-tool.py --del-tube-hv-profile NAME`

Perhaps with a disclaimer that the user has to agree to before writing the profile, some basic sanity checks for out-of-range values, and a limit to the number of saved profiles.

A bit of an ask to implement this, I'm sure, but I'm just throwing science at the wall and seeing what sticks, yknow?

**Comment by Gissio** [comment]
Rad Pro supports several devices, including some without a serial port, so your proposal is not feasible.

What makes more sense is to survey the FNIRSI GC-01 hardware versions and determine their optimum HV profiles. Are you willing to lead this effort?

**Comment by ThatBum** [comment]
> Rad Pro supports several devices, including some without a serial port, so your proposal is not feasible.

Right, forgot about that. :/ Thought it might be possible through the programming header but apparently not.

> What makes more sense is to survey the FNIRSI GC-01 hardware versions and determine their optimum HV profiles. Are you willing to lead this effort?

I mean I only have the one device so it's a single data point, but I'll do what I can. Like I said my 1G ohm resistors are still in the mail from Digikey so you'll have to wait.

I was under the impression that every device was a little different and needed its own tuning, kind of like overclocking. I guess devices with the same hardware versions are broadly similar enough for surveying them to be useful. Do your GC-01s not have M4011 tubes?

**Reply by Gissio** [reply]
It seems there are at least three hardware revisions of the GC-01.

What I meant is that you skim through the discussions of this forum and see what people reported for each hardware revision, or actively engage with the community and ask them for the best Accuracy and Energy-saving profile for their hardware.

It would also be important to know what hardware revisions go with what microcontroller (WCH or Geehy).

Then I can release a firmware version for every combination.

**Comment by ThatBum** [comment]
Reading through threads like [this one](https://github.com/Gissio/radpro/discussions/145) is pretty discouraging, it's just too inconsistent and unpredictable.

I did this [Google Image Search](https://www.google.com/search?q=fnirsi%20gc-01&hl=en&tbs=rimg:CQRhtdnq0UgjYQPNhUwbGlnbsgIAwAIA2AIB4AIA&udm=2&client=ms-android-att-us-rvc3&cs=1&sa=X&ved=0CBMQuIIBahcKEwiA05zD-piKAxUAAAAAHQAAAAAQSQ&biw=412&bih=826&dpr=3.5) for other units opened up and noticed that there are some with a resistor R38 in the voltage multiplier and ones without. Some have the pads but are unpopulated, some are just gone from the layout (the image from docs is one of these). Seems like this could occur in any combination of processor or tube too. Very confusing. You probably know all this already though...

There are even counterfeits of this model, can you even imagine?

I don't have an account on radmon.org or anywhere like that. I could try r/Radiation but I dunno if I'll get many people willing to do such a thing.

EDIT: Also think it's funny how before your firmware came along, people were modifying their GC-01s by running wires from the LED to the piezo buzzer just to get the click sound xD

**Comment by ThatBum** [comment]
Alright so here's some of my readings:

Factory default: 508 V
Energy-saving: 181 V

With the following custom settings I tried to tune the duty cycle for each frequency setting for highest voltage. I only have two decimal places on my multimeter in this range so this is the most precise I can get.

```
Frequency  Duty   Voltage
-------------------------
1.25 KHz   5%     248 V
2.5 KHz    5%     277 V*
2.5 KHz    10%    324 V
5 KHz      19%    408 V
10 KHz     35%    484 V
20 KHz     55%    534 V
40 KHz     50%    522 V
* This is the voltage where I start getting counts
```

Again I bring up this [dubious datasheet](https://github.com/LucasWolfgang/geiger_counter_v3/blob/master/M4011%20Geiger%20Tube%20Specification.pdf), the only one I can seem to find on this tube. According to this the operating voltage range is 380-450. Perhaps these would be good targets for the Energy-saving and Accuracy profiles? These are the settings on my device to hit these voltages.

380 V: 5 KHz, 12.75%
450 V: 10 KHz, 19.75%

I dunno if these settings are useful for anyone else's device but they may be more or less correct for devices with a Geehy CPU, M4011 tube, and R38 in place.

**Comment by dc1rdb** [comment]
My unit with Geehy CPU, J321 tube (equivalent to M4011), R38 was not populated by the factory:
Just soldered a 400 V zener ( two 200 V in series) on the R38 solder pads. With 5 kHz and 2.75 %, I now have a constant 390 V across the tube, independent of dose rate. Works perfect for me.

**Reply by ThatBum** [reply]
> My unit with Geehy CPU, J321 tube (equivalent to M4011), R38 was not populated by the factory: Just soldered a 400 V zener ( two 200 V in series) on the R38 solder pads. With 5 kHz and 2.75 %, I now have a constant 390 V across the tube, independent of dose rate. Works perfect for me.

Eyy, that's good to know. Answers the question I posed in [this post](https://github.com/Gissio/radpro/discussions/167). Would you mind taking a picture of your mod and posting it there?

It would seem that newer stock of GC-01s have M4011 tubes instead of J321s, and they have R38. If you can produce a higher output voltage with a lower duty cycle without R38, then in this configuration R38 it just wasting power, why is it there? Presumably to reduce the voltage for the J614/J613 versions.

**Reply by dc1rdb** [reply]
Here is a picture of my mod. I am sure this can be done better with smaller zeners, but I used what I had available in my component box.

![20241214_132137](https://github.com/user-attachments/assets/88a5dae4-1842-45a7-993e-01dcf667a302)

**Comment by ugopapetti** [comment]
Hi all,
please take care that the basic component for the HV production is the L1 coil.

Have a look here where there is a quick description: https://github.com/Gissio/radpro/discussions/145#discussioncomment-11506845

Net: are to be used frequencies up to a couple of kHz for "103" inductors and 10+ kHz for "102" inductors. 

From the electronic point of view, processor is not significant, neither the adopted tube (apart from the working voltage setting of course). 

After the frequency decision, the Duty Cycle will be used to adjust in the processor the HV / power. 

Of course a closed loop will be ideal for the control of the voltage. Applying a 400V zener diode (brute force regulation) solves, even if not the best solution, the price to be paid is the wasted energy used from the battery.

**Comment by dc1rdb** [comment]
Ugo, I am fully aware that a zener is not the perfect solution, compared to a regulated HV supply.
But it does a good job keeping the tube voltage within tube spec (better accuracy and no risk turning your geiger tube into a neon tube).
Additionally, with R38 replaced by a zener, a lower duty cycle is required to generate the correct HV, thereby reducing overall power consumption a bit. 
Again, it is not as good as a regulated supply would be, but a lot better than the factory build standard (at least for me).

**Comment by ThatBum** [comment]
I did this mod to my unit, works great. I put some heat shrink over the diodes so they didn't short out to the rest of the HV circuit, and I glopped it down with some silicone so vibrations and impacts don't put so much stress on the little SMD pads.
![PXL_20241220_232112493](https://github.com/user-attachments/assets/a9b3a9cb-0f13-4264-bfc8-a92ce5184f6d)
I looked around for smaller diodes but seems like 200V ones are either expensive, have long lead times, or both. So eh.

With the resistor gone I can get up to the correct voltage even with 1.5 KHz, but I'm using 10KHz at ugo's suggestion. I figure this would be more in the performance band of the inductor and not stress it as much. Duty cycle is set to 2.25%. Without the diodes (and without the resistor, to be clear) this was getting me about 450 V, and with the diodes it's a nice stable 380 V. I figure this is as good a set point as any. Gives it some headroom such that the voltage will remain constant even with a high count rate, but not so high as to waste excessive power. Either way, still much lower duty cycle required than with the resistor in place so I'm happy.

I also understand that this is sub-optimal compared to closed-loop control, but it's better than nothing and was easy to do. Also it seems to be the same implementation that the other meters that Rad Pro supports, so that's good enough for me.
